Signal/src/ViewControllers/BlockListUIUtils.m

```diff
@@ -62,6 +32,20 @@ + (void)showBlockPhoneNumberActionSheet:(NSString *)phoneNumber
                            completionBlock:completionBlock];
 }
 
++ (void)showBlockContactAccountActionSheet:(ContactAccount *)contactAccount
+                        fromViewController:(UIViewController *)fromViewController
+                           blockingManager:(OWSBlockingManager *)blockingManager
+                           contactsManager:(OWSContactsManager *)contactsManager
+                           completionBlock:(nullable BlockActionCompletionBlock)completionBlock
+{
+    NSString *displayName = [contactsManager displayNameForContactAccount:contactAccount];
```

Comment:
I haven't yet done the work to store the phone number "names/labels" for contacts with more than one account, e.g. "home" or "work".  That should be easy. I've made a card.